modules/mysql/mysql.go

```diff
@@ -0,0 +1,90 @@
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"strings"
+)
+
+const rootUser = "root"
+const defaultUser = "test"
+const defaultPassword = "test"
+const defaultDatabaseName = "test"
+
+// MySQLContainer represents the MySQL container type used in the module
+type MySQLContainer struct {
+	testcontainers.Container
+	username string
+	password string
+	database string
+}
+
+type PostgresContainerOption func(req *testcontainers.ContainerRequest)
+
+// StartContainer creates an instance of the MySQL container type
+func StartContainer(ctx context.Context, image string, opts ...PostgresContainerOption) (*MySQLContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        image,
+		ExposedPorts: []string{"3306/tcp", "33060/tcp"},
+		Env: map[string]string{
+			"MYSQL_USER":     defaultUser,
+			"MYSQL_PASSWORD": defaultPassword,
+			"MYSQL_DATABASE": defaultDatabaseName,
+		},
+		WaitingFor: wait.ForLog("port: 3306  MySQL Community Server"),
+	}
+
+	opts = append(opts, func(req *testcontainers.ContainerRequest) {
+		username := req.Env["MYSQL_USER"]
+		password := req.Env["MYSQL_PASSWORD"]
+		if strings.EqualFold(rootUser, username) {
+			delete(req.Env, "MYSQL_USER")
+		}
+		if len(password) != 0 && password != "" {
+			req.Env["MYSQL_ROOT_PASSWORD"] = password
+		} else if strings.EqualFold(rootUser, username) {
+			req.Env["MYSQL_ALLOW_EMPTY_PASSWORD"] = "yes"
+			delete(req.Env, "MYSQL_PASSWORD")
+		}
+	})
+
+	for _, opt := range opts {
+		opt(&req)
+	}
+
+	username, ok := req.Env["MYSQL_USER"]
+	if !ok {
+		username = rootUser
+	}
+	password := req.Env["MYSQL_PASSWORD"]
+
+	if len(password) == 0 && password == "" && !strings.EqualFold(rootUser, username) {
+		return nil, fmt.Errorf("empty password can be used only with the root user")
```

Comment:
We could define a custom and exported error here, something like 
```go
var ErrEmptyPasswForRoot = errors.New("empty password can be used only with the root user")
```
and finally in https://github.com/testcontainers/testcontainers-go/pull/943/files#diff-02c2cf13c522643c66ea754bfc71eeaaec57856bf6abdb78f50a93205f429325R65 check for the error with `errors.Is` https://pkg.go.dev/errors#Is

Reply:
But I'm not sure about this, as it would mean adding Error types for each module to the public API, which although interesting at some point, could come with different maintainability concerns, like naming, which error types support, etc